engine/storage_mongo_datadb.go

```diff
@@ -1148,8 +1148,11 @@ func (ms *MongoStorage) SetAccount(acc *Account) error {
 func (ms *MongoStorage) RemoveAccount(key string) error {
 	session, col := ms.conn(colAcc)
 	defer session.Close()
-	return col.Remove(bson.M{"id": key})
-
+	err := col.Remove(bson.M{"id": key})
```

Comment:
Please add the err in the return signature and compare it with a single expression, ie: 
if err = col.Remove;...; err == mgro {err = utils.ErrNotFound}